### ISSUE TITLE
How to use aut0-lock with requirejs?

### ISSUE THREAD

**Issue by afgallo** [auth0/lock#172]
Hi,

I'd like to use auth0-lock with requirejs. Below is my current setup. I basically have my requirejs config and AMD module that requires auth0-lock, I cannot create an instance of Auth0Lock because when I do a console.log as shown below, I get the output of Sizzle.

Any ideas? What am I doing wrong?

```javascript
// requirejs config
paths: {
    'auth0-lock': '../bower/auth0-lock/build/auth0-lock.min'
}

// my AMD module definition
define(['jquery', 'knockout', 'text!./sign-in.html', 'auth0-lock'], function ($, ko, template, auth) {
    console.log(auth);
    console.log(new auth());
}

// the result of console.log(auth);
function Sizzle(selector,context,results,seed){var m,i,elem,nid,match,groups,newSelector,newContext=context&&context.ownerDocument,nodeType=context?context.nodeType:9;results=results||[];if(typeof selector!=="string"||!selector||nodeType!==1&&nodeType!==9&&nodeType!==11){return results}if(!seed){if((context?context.ownerDocument||context:preferredDoc)!==document){setDocument(context)}context=context||document;if(documentIsHTML){if(nodeType!==11&&(match=rquickExpr.exec(selector))){if(m=match[1]){if(nodeType===9){if(elem=context.getElementById(m)){if(elem.id===m){results.push(elem);return results}}else{return results}}else{if(newContext&&(elem=newContext.getElementById(m))&&contains(context,elem)&&elem.id===m){results.push(elem);return results}}}else if(match[2]){push.apply(results,context.getElementsByTagName(selector));return results}else if((m=match[3])&&support.getElementsByClassName&&context.getElementsByClassName){push.apply(results,context.getElementsByClassName(m));return results}}if(support.qsa&&!compilerCache[selector+" "]&&(!rbuggyQSA||!rbuggyQSA.test(selector))){if(nodeType!==1){newContext=context;newSelector=selector}else if(context.nodeName.toLowerCase()!=="object"){if(nid=context.getAttribute("id")){nid=nid.replace(rescape,"\\$&")}else{context.setAttribute("id",nid=expando)}groups=tokenize(selector);i=groups.length;while(i--){groups[i]="[id='"+nid+"'] "+toSelector(groups[i])}newSelector=groups.join(",");newContext=rsibling.test(selector)&&testContext(context.parentNode)||context}if(newSelector){try{push.apply(results,newContext.querySelectorAll(newSelector));return results}catch(qsaError){}finally{if(nid===expando){context.removeAttribute("id")}}}}}}return select(selector.replace(rtrim,"$1"),context,results,seed)}

// the result of console.log(new auth());
[]

**Comment by cristiandouce** [comment]
Would you have an example or seed to test that out?
Could you also try requiring auth0-lock first?
Thanks!

**Comment by afgallo** [comment]
@cristiandouce requiring it first didn't work ! I'll work on an seed and send you asap.

**Comment by cristiandouce** [comment]
@afgallo thanks!

**Comment by afgallo** [comment]
There you go @cristiandouce:

https://github.com/afgallo/auth0lock-requirejs

**Comment by cristiandouce** [comment]
Thanks @afgallo! I'll take a look at that later and ping you back when find a solution.

**Comment by afgallo** [comment]
@cristiandouce any news?

**Comment by cristiandouce** [comment]
@afgallo nope, not yet. I'm not requirejs expert. But I've found some odd behaviors when there are "anonymous" defined "dependencies" (like Sizzle in this case).

http://requirejs.org/docs/errors.html#mismatch

**Comment by cristiandouce** [comment]
For instance, just to prove the concept I gave a `name` property to all `define(definition)` calls inside dependencies and also `jquery`'s define for `sizzle`.

This was the result:
![image](https://cloud.githubusercontent.com/assets/367429/7707886/b0d281f2-fe28-11e4-8de0-fe2882d394c6.png)

`Auth0Lock` does define itself with a name to avoid this kind of issues. But, I'm not yet sure how to avoid/sort this issue for dependencies of dependencies when there are `anonymous` definitions.

**Comment by jfromaniello** [comment]
Fixed on version 7.5.4.

Thanks for reporting it!

**Comment by cristiandouce** [comment]
@jose :rose:
